assignments/homework/Raymond Ingram/Homework 2.py

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # <section class="challenge panel panel-success">
+# <div class="panel-heading">
+# <h2><span class="fa fa-pencil"></span> Challenge</h2>
+# </div>
+# 
+# 
+# <div class="panel-body">
+# 
+# <ol>
+# <li>Examine the headers of the first and second HDU in the point source catalog and try adding new keywords to them or changing them</li>
+# <li>Make a histogram of the G-band magnitude of the sources in the catalog - can you figure out what the upper limit for Gmag was when the table was selected?</li>
+# <li>Make a plot of the position of the sources on the sky in the point source catalog, in Galactic coordinates l and b (you will need to use what we learned about SkyCoord in the previous tutorial)</li>
+# <li>Try and produce a new FITS file that contains the image in the primary HDU and the above table in the second HDU</li>
+# </ol>
+# 
+# </div>
+# 
+# </section>
+# 
+
+# In[1]:
+
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+import matplotlib.pyplot as plt
+plt.rc('image', origin='lower')
+plt.rc('figure', figsize=(10, 6))
+
+
+# In[2]:
+
+
+import astropy.units as u
+import numpy as np
+from astropy.io import fits
+from astropy.coordinates import *
+from astropy import visualization
+from astropy.table import Table
+
+
+# In[3]:
+
+
+hdulist1 = fits.open('../../../astropy_notebooks/data/LMCDensFits1k.fits')
+hdulist2 = fits.open('../../../astropy_notebooks/data/gaia_lmc_psc.fits')
+
+
+# In[4]:
+
+
+hdulist1[0].header
+
+
+# In[5]:
+
+
+hdulist2[1].header
+
+
+# In[6]:
+
+
+hdulist2[1].header['BITPIX'] = 'YOLO'
+hdulist2[1].header['BLARG'] = 'does this work?'
+hdulist2[1].header
+
+
+# In[7]:
+
+
+hdu2 = Table.read('data/gaia_lmc_psc.fits')
+hdu2
+
+
+# In[8]:
+
+
+_ = plt.hist(hdu2['phot_g_mean_mag'], bins=100)
+
+
+# In[9]:
+
+
+coord1 = SkyCoord(hdu2['ra']*u.deg, hdu2['dec']*u.deg)
+coord2 = coord1.transform_to('galactic')
+coord2
+plt.scatter(coord2.l,coord2.b)
+
+
+# In[10]:
+
+
+hdu = fits.PrimaryHDU()
+hdu
+
+
+# In[11]:
+
+
+hdu.table = hdu2
+dataz = fits.getdata('data/gaia_lmc_psc.fits')
+hdu2.data = dataz
```

Comment:
The exercise was to create a hdulist with multiple HDUs, one from the image the other from the table.

See e.g.:
```
hdulist_combined = fits.HDUList([hdulist1[0], hdulist2[1]])
hdulist_combined.writeto('newFits1.fits', overwrite=True)
```